presto-kudu/src/test/java/io/prestosql/plugin/kudu/TestKuduIntegrationGroupedExecution.java

```diff
@@ -0,0 +1,207 @@
+import static io.prestosql.SystemSessionProperties.DYNAMIC_SCHEDULE_FOR_GROUPED_EXECUTION;
+import static io.prestosql.SystemSessionProperties.ENABLE_DYNAMIC_FILTERING;
+import static io.prestosql.SystemSessionProperties.GROUPED_EXECUTION;
+import static io.prestosql.plugin.kudu.KuduQueryRunnerFactory.createKuduQueryRunner;
+import static io.prestosql.sql.planner.optimizations.PlanNodeSearcher.searchFrom;
+import static io.prestosql.sql.planner.planprinter.PlanPrinter.textLogicalPlan;
+import static java.lang.String.format;
+
+public class TestKuduIntegrationGroupedExecution
+        extends AbstractTestQueryFramework
+{
+    public static final String KUDU_GROUPED_EXECUTION = "kudu.grouped_execution";
+    private TestingKuduServer kuduServer;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        kuduServer = new TestingKuduServer();
+        return createKuduQueryRunner(kuduServer, "test_grouped_execution");
+    }
+
+    @AfterClass(alwaysRun = true)
+    public final void destroy()
+    {
+        kuduServer.close();
+    }
+
+    @Test
+    public void testGroupedExecutionJoin()
+    {
+        Session groupedExecutionSessionProperties = Session.builder(getSession())
+                .setSystemProperty(COLOCATED_JOIN, "true")
+                .setSystemProperty(GROUPED_EXECUTION, "true")
+                .setSystemProperty(CONCURRENT_LIFESPANS_PER_NODE, "1")
+                .setSystemProperty(DYNAMIC_SCHEDULE_FOR_GROUPED_EXECUTION, "false")
+                .setSystemProperty(ENABLE_DYNAMIC_FILTERING, "false")
+                .setSystemProperty(KUDU_GROUPED_EXECUTION, "true")
+                .build();
+
+        assertUpdate("CREATE TABLE IF NOT EXISTS test_grouped_execution_t1 (" +
+                "key1 INT WITH (primary_key=true), " +
+                "key2 INT WITH (primary_key=true), " +
+                "attr1 INT" +
+                ") WITH (" +
+                " partition_by_hash_columns = ARRAY['key1'], " +
+                " partition_by_hash_buckets = 2" +
+                ")");
+
+        assertUpdate("CREATE TABLE IF NOT EXISTS test_grouped_execution_t2 (" +
+                "key1 INT WITH (primary_key=true), " +
+                "key2 INT WITH (primary_key=true), " +
+                "attr2 decimal(10, 6)" +
+                ") WITH (" +
+                " partition_by_hash_columns = ARRAY['key1'], " +
+                " partition_by_hash_buckets = 2" +
+                ")");
+
+        assertUpdate("INSERT INTO test_grouped_execution_t1 VALUES (0, 0, 0), (0, 1, 0), (1, 1, 1)", 3);
+        assertUpdate("INSERT INTO test_grouped_execution_t2 VALUES (0, 0, 0), (1, 1, 1), (1, 2, 1)", 3);
+        assertQuery(groupedExecutionSessionProperties, "SELECT t1.* FROM test_grouped_execution_t1 t1 join test_grouped_execution_t2 t2 on t1.key1=t2.key1 WHERE t1.attr1=0", "SELECT * FROM (VALUES (0, 0, 0), (0, 1, 0))", assertRemoteExchangesCount(1));
```

Comment:
How did you verify grouped execution actually worked? This assertion will pass disregarding if it is enabled or not.

Reply:
Well, if grouped execution worked, `assertRemoteExchangesCount` will assert 1, or it will be > 1. This is verified.   :)